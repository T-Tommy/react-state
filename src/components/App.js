import React, { PureComponent } from 'react';
import RandomColor from './RandomColor';
import Button from './Button';
import styles from './App.css';

export default class App extends PureComponent {
  state = {
    blocks: 1
  }
  addBlock = () => this.setState(state => ({ blocks: state.blocks + 1 }))
  render() {
    return (
      <>
        <Button method={ this.addBlock } />
        <section className={ styles.app }>
          {
            [...Array(this.state.blocks)].map((_, i) =>
              <RandomColor key={ i } />
            )
          }
        </section>
      </>
    );
  }
}
