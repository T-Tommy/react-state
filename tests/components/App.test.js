import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/components/App';

describe('App component', () => {
  it('renders the App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('adds a block', () => {
    const wrapper = shallow(<App />);
    wrapper.instance().addBlock();
    expect(wrapper.state('blocks')).toEqual(2);
  });
});
