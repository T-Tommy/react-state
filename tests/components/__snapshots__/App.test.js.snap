// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the App 1`] = `
<Fragment>
  <Button
    method={[Function]}
  />
  <section
    className="app"
  >
    <RandomColor
      key="0"
    />
  </section>
</Fragment>
`;
